### ISSUE TITLE
Inference issue

### ISSUE THREAD

**Issue by ezmonyi** [Zyphra/Zonos#221]
I am having a what seems like model compute level issue, it's quite long so I will just attach the last part as screenshot.

        # File: /home/yi/projects/tts/zonos/zonos/model.py:101 in <listcomp>, code: return torch.stack([head(hidden_states) for head in self.heads], dim=1)
        linear_104: "bf16[2, 1, 1026][1026, 1026, 1]cuda:0" = torch._C._nn.linear(last_hidden_states, l_self_modules_heads_modules_0_parameters_weight_, None);  l_self_modules_heads_modules_0_parameters_weight_ = None
        linear_105: "bf16[2, 1, 1026][1026, 1026, 1]cuda:0" = torch._C._nn.linear(last_hidden_states, l_self_modules_heads_modules_1_parameters_weight_, None);  l_self_modules_heads_modules_1_parameters_weight_ = None
        linear_106: "bf16[2, 1, 1026][1026, 1026, 1]cuda:0" = torch._C._nn.linear(last_hidden_states, l_self_modules_heads_modules_2_parameters_weight_, None);  l_self_modules_heads_modules_2_parameters_weight_ = None
        linear_107: "bf16[2, 1, 1026][1026, 1026, 1]cuda:0" = torch._C._nn.linear(last_hidden_states, l_self_modules_heads_modules_3_parameters_weight_, None);  l_self_modules_heads_modules_3_parameters_weight_ = None
        linear_108: "bf16[2, 1, 1026][1026, 1026, 1]cuda:0" = torch._C._nn.linear(last_hidden_states, l_self_modules_heads_modules_4_parameters_weight_, None);  l_self_modules_heads_modules_4_parameters_weight_ = None
        linear_109: "bf16[2, 1, 1026][1026, 1026, 1]cuda:0" = torch._C._nn.linear(last_hidden_states, l_self_modules_heads_modules_5_parameters_weight_, None);  l_self_modules_heads_modules_5_parameters_weight_ = None
        linear_110: "bf16[2, 1, 1026][1026, 1026, 1]cuda:0" = torch._C._nn.linear(last_hidden_states, l_self_modules_heads_modules_6_parameters_weight_, None);  l_self_modules_heads_modules_6_parameters_weight_ = None
        linear_111: "bf16[2, 1, 1026][1026, 1026, 1]cuda:0" = torch._C._nn.linear(last_hidden_states, l_self_modules_heads_modules_7_parameters_weight_, None);  l_self_modules_heads_modules_7_parameters_weight_ = None
        linear_112: "bf16[2, 1, 1026][1026, 1026, 1]cuda:0" = torch._C._nn.linear(last_hidden_states, l_self_modules_heads_modules_8_parameters_weight_, None);  last_hidden_states = l_self_modules_heads_modules_8_parameters_weight_ = None
        
         # File: /home/yi/projects/tts/zonos/zonos/model.py:101 in apply_heads, code: return torch.stack([head(hidden_states) for head in self.heads], dim=1)
        stack_52: "bf16[2, 9, 1, 1026][9234, 1026, 1026, 1]cuda:0" = torch.stack([linear_104, linear_105, linear_106, linear_107, linear_108, linear_109, linear_110, linear_111, linear_112], dim = 1);  linear_104 = linear_105 = linear_106 = linear_107 = linear_108 = linear_109 = linear_110 = linear_111 = linear_112 = None
        
         # File: /home/yi/projects/tts/zonos/zonos/model.py:111 in _compute_logits, code: logits = self.apply_heads(last_hidden_states).squeeze(2).float()
        squeeze: "bf16[2, 9, 1026][9234, 1026, 1]cuda:0" = stack_52.squeeze(2);  stack_52 = None
        logits: "f32[2, 9, 1026][9234, 1026, 1]cuda:0" = squeeze.float();  squeeze = logits = None
        
         # File: /home/yi/projects/tts/zonos/zonos/model.py:112 in _compute_logits, code: if cfg_scale != 1.0:
        ne: "b8[][]cpu" = l_cfg_scale_ != 1.0;  l_cfg_scale_ = ne = None

<img width="1470" alt="Image" src="https://github.com/user-attachments/assets/018ca2af-324c-46e1-aad6-87d52446d0a0" />

OS: 
Distributor ID: Ubuntu
Description:    Ubuntu 22.04.2 LTS
Release:        22.04
Codename:       jammy

GPU: 

<img width="652" alt="Image" src="https://github.com/user-attachments/assets/94a975fe-3ae7-4513-8bf7-f3b4c37873ba" />
